### ISSUE TITLE
Add a report button to community discussions and comments

### ISSUE THREAD

**Issue by jesseallhands** [Couchers-org/couchers#5122]
**Problem:**
There is currently no way to report a discussion OR comment in a community.

**Solution:**
Add a report icon (flag icon) to discussions and comments (similar to the messaging system). Same symbol, same behavior.

We need to add the flag symbol here for discussions (similar placement on mobile):
![image](https://github.com/user-attachments/assets/62cf54fe-c3f9-42a6-b237-48c331eb9cf2)

And here for comments:
Desktop
![image](https://github.com/user-attachments/assets/e8a57a19-e0ca-4470-bf87-e9c03db82a85)

Mobile:
![image](https://github.com/user-attachments/assets/b859cf62-2139-47b1-9a02-7eb3d4f3623b)

Here is what the current report icon looks like on messages:
![Screenshot_20241111_082400_Firefox](https://github.com/user-attachments/assets/f02e7a28-60df-4115-b511-7993ccfabb5a)

I'm not sure if the current content reporting API is able to include the ID of the discussion or comment, so the content reporting api may need to be updated. But other than that this is mostly a frontend issue.

**Comment by bakeiro** [comment]
we just add a 'report' button? maybe isn't to too visible seeing a big report button for each comment?

[I say this because if it's as big as the reply button I feel it's too big/visible... maybe we should wait for an UI/UX mockup first?]

**Comment by bakeiro** [comment]
why not something like this? @jesseallhands 
![Image](https://github.com/user-attachments/assets/88b2c394-89da-408b-b207-de6b5c73192e)

**Comment by bakeiro** [comment]
U see what I mean?
![Image](https://github.com/user-attachments/assets/5d00750f-e86d-417b-a09f-6bca3d591a57)

**Comment by bakeiro** [comment]
maybe we move this to 'must have' until we have a decision? [I feel like urgent are for tickets that we know exactly what to do and we need it ASAP]

**Comment by jesseallhands** [comment]
I mean to use the same report feature we currently have. It's not technically a button, it's a clickable icon:  
![Screenshot_20241111_082400_Firefox](https://github.com/user-attachments/assets/71c369e5-0c47-4f54-a856-9dc6852f7df0)

**Comment by mayTkhin24** [comment]
Hi @jesseallhands,

We are a team of five students who would like to contribute to this issue. We plan to spend the next two weeks familiarizing ourselves with the codebase and working on this issue.

We would appreciate any guidance to help us get started.

Looking forward to your response!

**Comment by jesseallhands** [comment]
@mayTkhin24 thanks for reaching out! If you have any UI/UX or design questions, I can help. There should be some docs to help you get things set up, but if you have any questions getting set up @aapeliv or @nabramow can help you.

**Comment by mayTkhin24** [comment]
@jesseallhands  Thank you for your reply!

@aapeliv @nabramow We've managed to set it up locally, but we wanted to get some more information on this task. Is it possible to set up a meeting? :)

**Comment by nabramow** [comment]
> @aapeliv @nabramow We've managed to set it up locally, but we wanted to get some more information on this task. Is it possible to set up a meeting? :)

Hi @mayTkhin24 - probably best to ask the specific questions you need clarification on here, or come to our weekly meeting on Tuesday at 9pm UTC.

First thing I'd do get the app up and running and find the places the flag icons should go, find the existing flag icon, and try to identify the backend function you'll connect to.

@aapeliv Do you know if the backend is done for this? Or is that part of the ticket?

**Comment by aapeliv** [comment]
Yes backend is done and ready. I talked with @mayTkhin24 today on zoom. I'll message you on slack @nabramow .

**Comment by aapeliv** [comment]
CMU crowd is working on this.
